src/Utilities/src/MoistThermodynamics.jl

```diff
@@ -411,10 +413,17 @@ The optional input value of the temperature `T_init` is taken as the initial
 value of the saturation adjustment iterations.
 """
 function saturation_adjustment(E_int, ρ, q_t, T_init = T_triple)
-
-    T = find_zero(x -> internal_energy_sat(x, ρ, q_t) - E_int, T_init,
-            Order1(), atol=1e-3*cv_d)
-
+    tol_abs = 1e-3*cv_d
+    iter_max = 100
+    args = ()
+    T_0, T_1 = T_init, T_init*1.01
```

Comment:
I'm working on this now. I think that I've got an idea of how to address this and #41 in a pretty simple and encapsulated way.

Reply:
It turns out that allowing the function to operate on single `AbstractFloat`'s and broadcast over `Array`'s, or `CuArray`'s, is difficult because of the type system hierarchy (a figure of the CPU types are here: https://en.wikibooks.org/wiki/Introducing_Julia/Types#/), and the CuArray type structure is built from the top down:

```
julia> supertype(CuArray)
GPUArrays.GPUArray
julia> supertype(supertype(CuArray))
DenseArray
julia> supertype(supertype(supertype(CuArray)))
AbstractArray
```